### PR TITLE
feat[docs]: add bug bounty program to security policy

### PR DESCRIPTION
**Pull request by cyberthirst**
### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Supported Versions
 
 - it is recommended to follow the list of known [vulnerabilities](https://github.com/vyperlang/vyper/security/advisories) and stay up-to-date with the latest releases
-  - as of May 2024, the `0.4.0` release is the most secure and the most comprehensively reviewed one and is recommended for use in production environments
+  - as of May 2024, the `0.4.0` release is the most comprehensively reviewed one and is recommended for use in production environments
```
Comment (pcaversaccio) [inline SECURITY.md:6]:
```suggestion
  - as of May 2024, the [`0.4.0`](https://github.com/vyperlang/vyper/releases/tag/v0.4.0) release is the most comprehensively reviewed one and is recommended for use in production environments
```

File: SECURITY.md [hunk 2]
```diff
@@ -26,7 +26,20 @@ we will add an entry to the list of security advisories for posterity and refere
 
 
 ## Bug Bounty Program
-- as of May 2024, Vyper does not have a bug bounty program. It is planned to instantiate one soon.
+- Vyper runs a bug bounty program via Ethereum Foundation.
```
Comment (pcaversaccio) [inline SECURITY.md:29]:
```suggestion
- Vyper runs a bug bounty program via the Ethereum Foundation.
```

File: SECURITY.md [hunk 3]
```diff
@@ -26,7 +26,20 @@ we will add an entry to the list of security advisories for posterity and refere
...
+  - Bugs should be reported through the [Ethereum Foundation's bounty program](https://bounty.ethereum.org/).
```
Comment (pcaversaccio) [inline SECURITY.md:30]:
```suggestion
  - Bugs should be reported through the [Ethereum Foundation's bounty program](https://ethereum.org/bug-bounty).
```

File: SECURITY.md [hunk 4]
```diff
@@ -26,7 +26,20 @@ we will add an entry to the list of security advisories for posterity and refere
...
+
+### Scope
+- Rules from the Ethereum Foundation's bug bounty program apply. Here we further clarify the scope of the Vyper bounty program.
```
Comment (pcaversaccio) [inline SECURITY.md:33]:
```suggestion
- Rules from the Ethereum Foundation's bug bounty program apply; for any questions please reach out [here](mailto:bounty@ethereum.org). Here we further clarify the scope of the Vyper bounty program.
```

File: SECURITY.md [hunk 5]
```diff
@@ -26,7 +26,20 @@ we will add an entry to the list of security advisories for posterity and refere
...
+- If a compiler bug affects production code, it is in scope.
+  - This includes bugs in older compiler versions still used in production (excluding known issues).
+- If a compiler bug does not currently affect production but is likely to in the future, it is in scope.
+  - This mainly applies to the latest compiler release (e.g., a new release is available but contracts are not yet deployed with it).
```
Comment (pcaversaccio) [inline SECURITY.md:37]: there might be questions re what "new" exactly means (like 2 weeks, 6 months etc.). But we can address this if someone challenges it.

File: SECURITY.md [hunk 6]
```diff
@@ -26,7 +26,20 @@ we will add an entry to the list of security advisories for posterity and refere
...
+  - Experimental features are out of scope, as they are not intended for production and are unlikely to affect production code.
```
Comment (pcaversaccio) [inline SECURITY.md:38]:
```suggestion
  - Experimental features (e.g. `--experimental-codegen`) are out of scope, as they are not intended for production and are unlikely to affect production code.
```

**Review by pcaversaccio — APPROVED** [review]
Some nit comments; overall good to go IMO.

File: SECURITY.md (hunk 5 above)
Comment (cyberthirst) [reply SECURITY.md:37]: Yeah, I agree, but it was intentional. I think it's realistic that 0.3.10 might be used, but how far into the past should we go? I leave it up to the hunters to convince us / ask further questions.

Comment (pcaversaccio) [reply SECURITY.md:37]: I think `0.3.7` has been used widely but we shouldn't overthink it too much indeed.

File: SECURITY.md [hunk 7]
```diff
@@ -26,7 +26,20 @@ we will add an entry to the list of security advisories for posterity and refere
...
+- If a compiler bug affects production code, it is in scope.
+  - This includes bugs in older compiler versions still used in production (excluding known issues).
+- If a compiler bug does not currently affect production but is likely to in the future, it is in scope.
+  - This mainly applies to the latest compiler release (e.g., a new release is available but contracts are not yet deployed with it).
+  - Experimental features (e.g. `--experimental-codegen`) are out of scope, as they are not intended for production and are unlikely to affect production code.
+  - Bugs in older compiler versions are generally out of scope, as they are no longer used for new contracts.
+- If a vulnerability affects multiple contracts, the whitehat is eligible for only one payout (though the severity of the bug may increase).
+- Eligibility for project-specific bounties is independent of this bounty.
+- [Known issues](https://github.com/vyperlang/vyper/security/advisories) are not eligible for the bounty program, as they are publicly disclosed and protocols should structure their contracts accordingly.
```
[the remark this answers is not in the document]
Comment (cyberthirst) [reply SECURITY.md]: addressed in https://github.com/vyperlang/vyper/pull/4230/commits/6f8e66b773a3195a5ae544725a281f8f5e9435a4

**Review by charles-cooper — APPROVED** [review]

**Pull request merged**